### DISCUSSION TITLE
Custom Avatars Not Orienting Properly During Motion.

### DISCUSSION THREAD

**Discussion by NovySan** [mozilla/hubs#3874] [Avatar Questions] [answered]
Our custom avatar, literally just happy blobs with faces, enter a Room facing one direction (the direction they were facing in Blender) but never rotate or orient themselves to the direction of travel one would expect when moving. Should I expect them to? Non-custom avatars seem to orient themselves toward the direction of travel.

CAVEAT: I only know this because I'm placing a camera and can tell that I'm not oriented the way I should be facing after moving.

**Comment by j-conrad** [comment, marked as answer]
When you create an avatar using Blender, you'll need the avatar's 'front' to face toward the negative Y direction. This will ensure that it will face the 'right' way in Hubs. In Blender, negative Y is designated by the viewport gizmo's green dot with no line.
![image](https://user-images.githubusercontent.com/31968496/107310793-f9206380-6a41-11eb-96b0-66e9a0a64053.png)

The other thing to make sure of before you export to .glb is that your model's transformations (position/rotation/scale) are all reset. In other words, Position should read 0,0,0, Rotation 0,0,0, and Scale 1,1,1.
If they are not, you can select the offending meshes and type `CTRL+a`, then select whichever items need resetting.

**Comment by NovySan** [comment]
Good to know. I gave up using custom avatars since I didn't have time to
troubleshoot but I'll visit this again later. I don't have any rigging in
my model, it's literally just a deformed sphere with a texture map on it
to make a "blobby"-like character.

**Comment by evhan55** [comment]
Thanks for this! CrimsonZA on Discord was able to help me and told me the model has to be at 0,0,0 (not just applied there) in order to rotate correctly
